src/hazmat/lucas.rs

```diff
@@ -172,15 +164,18 @@ impl LucasBase for BruteForceBase {
 }
 
 /// For the given odd `n`, finds `s` and odd `d` such that `n + 1 == 2^s * d`.
-fn decompose<const L: usize>(n: &Uint<L>) -> (usize, Uint<L>) {
+fn decompose<T: UintLike>(n: &T) -> (u32, T) {
     debug_assert!(bool::from(n.is_odd()));
 
     // Need to be careful here since `n + 1` can overflow.
     // Instead of adding 1 and counting trailing 0s, we count trailing ones on the original `n`.
 
     let s = n.trailing_ones();
     // This won't overflow since the original `n` was odd, so we right-shifted at least once.
-    let d = Option::from((n >> s).checked_add(&Uint::<L>::ONE)).expect("Integer overflow");
+    // TODO: shr(s-1).shr(1) is a hack around the fact that a full right shift will panic
+    // see https://github.com/RustCrypto/crypto-bigint/commit/55312b6aa71#r134960147
+    let d = Option::from((n.clone().shr(s - 1).shr(1)).checked_add(&T::one()))
```

Comment:
This whole thing can be vartime, so you can just use `shr_vartime(s).unwrap_or(T::zero())`